src/components/UNSTABLE__Pagination/PageInput.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * @file Page input.
+ * @copyright IBM Security 2020
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import setupGetInstanceId from 'carbon-components-react/lib/tools/setupGetInstanceId';
+import NumberInput from '../NumberInput';
+import { appendComponentNamespace } from '../../globals/namespace/index';
+import { namespace as paginationNamespace } from './Pagination';
+
+const namespace = appendComponentNamespace(paginationNamespace, 'page-input');
+const getInstanceId = setupGetInstanceId();
+
+function PageInput({
+  className,
+  currentPage,
+  id,
+  invalidText,
+  label,
+  max,
+  min,
+  totalPages,
+  ...other
+}) {
+  const instanceId = `${namespace}__input-${getInstanceId()}`;
+
+  return (
+    <NumberInput
+      className={classnames(namespace, className)}
+      hideLabel
+      id={instanceId || id}
```

Comment:
I opted to do this for both the `PageInput` and `PageSelector` because they just have one place where `id` is used & it was easy enough to generate and use an instance ID that could be replaced.

However, I opened not to do this for the parent pagination component because that component ends up using an id throughout for setting a11y attributes etc. I didn't get to make the default of that parent component's `id` equal to `getInstanceId()` or anything like that. 🤔 Seemed... strange to do so? 

But let me know if you disagree / have concerns 👍